File: src/Web/TypeScriptTS/facade.ts
class AuthService {
  authenticate(user: string): string {
    return `auth(${user})`;
  }
}

class InventoryService {
  reserve(sku: string): string {
    return `reserve(${sku})`;
  }
}

class BillingService {
  charge(cents: number): string {
    return `charge(${cents})`;
  }
}

class CheckoutFacade {
  constructor(
    private readonly auth: AuthService,
    private readonly inventory: InventoryService,
    private readonly billing: BillingService,
  ) {}

  checkout(user: string, sku: string, cents: number): string {
    return [
      this.auth.authenticate(user),
      this.inventory.reserve(sku),
      this.billing.charge(cents),
    ].join(">");
  }
}

const facade = new CheckoutFacade(
  new AuthService(),
  new InventoryService(),
  new BillingService(),
);

console.log(`checkout=${facade.checkout("alice", "SKU-42", 499)}`);

File: src/Other/MicroPython/facade.py
class AuthService:
    def authenticate(self, user):
        return "auth(%s)" % user


class InventoryService:
    def reserve(self, sku):
        return "reserve(%s)" % sku


class BillingService:
    def charge(self, cents):
        return "charge(%d)" % cents


class CheckoutFacade:
    def __init__(self, auth, inventory, billing):
        self.auth = auth
        self.inventory = inventory
        self.billing = billing

    def checkout(self, user, sku, cents):
        return "%s>%s>%s" % (
            self.auth.authenticate(user),
            self.inventory.reserve(sku),
            self.billing.charge(cents),
        )


facade = CheckoutFacade(AuthService(), InventoryService(), BillingService())
print("checkout=" + facade.checkout("alice", "SKU-42", 499))
